web/src/lib/stores/assets-store.svelte.ts

```diff
@@ -57,6 +57,13 @@ function updateObject(target: any, source: any): boolean {
   return updated;
 }
 
+export function assetSnapshot(asset: AssetResponseDto) {
```

Comment:
Can you help explain the reason behind this method/mechanism?

Reply:
One problem with the assetInteraction class (left-over from before asset-store used lived-objects) was that it used asset-store model objects directly. Those objects are live objects. Sometimes actions modify these assets "helpfully" - which causes some downstream issues. The fix is to take a snapshot of those objects before giving them to any selection manager. However, because a snapshot is taken every time assets are accessed, and they are sometimes accessed multiple times, the Set (which acts on object identity) sees two snaps of the same object as 2 objects. Fix for that is to use an array, and not use object identity, rather assetId.